### PR TITLE
JBPM-6770: Information not available in TaskLifeCycleEventListener

### PR DESCRIPTION
**Pull request by mcivantos-tribalyte**
Added message to the TASKEVENT with people assigment in case of forward event

### CODE REVIEW

**Comment by mcivantos-tribalyte** [comment]
@mswiderski  Test added.

**Comment by mswiderski** [comment]
ok to test

**Comment by tsurdilo** [comment]
Jenkins retest please

File: jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java [hunk 1]
```diff
@@ -1328,6 +1327,33 @@ public void testUpdateTaskContentEvents() {
...
+                .addPotentialGroup("Knights Templer")
+                .setAdminUser("Administrator")
+                .getTask();
+
+        taskService.addTask(task, new HashMap<String, Object>());
+        long taskId = task.getId();
+
+        assertAuditTaskInfoGroup("Knights Templer", "Ready", taskId);
+
+        taskService.forward(taskId, "Administrator", "Jon Snow");
```
Comment (domhanak) [inline jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: I do not see user with name "Jon Snow" defined anywhere. I think it is the reason the test fails

File: jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java [hunk 2]
```diff
@@ -1328,6 +1327,33 @@ public void testUpdateTaskContentEvents() {
...
+        List<TaskEvent> taskEvents = taskAuditService.getAllTaskEvents(taskId, new QueryFilter());
+       
+        Assertions.assertThat(taskEvents).hasSize(2);
+        
+        Assertions.assertThat(taskEvents.get(0).getType()).isEqualTo(TaskEventType.ADDED);
+        Assertions.assertThat(taskEvents.get(0).getUserId()).isNull();
+       
+        Assertions.assertThat(taskEvents.get(1).getType()).isEqualTo(TaskEventType.FORWARDED);
+        Assertions.assertThat(taskEvents.get(1).getUserId()).isEqualTo("Administrator");        
+        assertNotNull(taskEvents.get(1).getMessage());
```
Comment (domhanak) [inline jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: > assertThat(taskEvents.get(1).getMessage()).isNotNull()

File: jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java [hunk 3]
```diff
@@ -378,9 +378,17 @@ public void afterTaskForwardedEvent(TaskEvent event) {
         Task ti = event.getTask();
         TaskPersistenceContext persistenceContext = getPersistenceContext(((TaskContext)event.getTaskContext()).getPersistenceContext());
         try {
-            persistenceContext.persist(new TaskEventImpl(ti.getId(), org.kie.internal.task.api.model.TaskEvent.TaskEventType.FORWARDED, ti.getTaskData().getProcessInstanceId(), ti.getTaskData().getWorkItemId(), userId));
-    
-    
+            StringBuilder message = new StringBuilder();
+            String entitiesAsString = (ti.getPeopleAssignments().getPotentialOwners()).stream().map(oe -> oe.getId()).collect(Collectors.joining(","));
+            message.append("Forward to [" + entitiesAsString + "]");
```
Comment (domhanak) [inline jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java]: Usage of _StringBuilder_  can be replaced by simple _String_  here. It looks to me that StringBuilder is an overkill, but take it as a suggestion.

For example:
String message = String.format("Forward to [%s]", entitiesAsString);

Comment (mcivantos-tribalyte) [reply jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java]: The number of entities into getPotentialOwners could be pretty big. Anyway, to be coherence, the rest of messages in other methods of this class are defined like StringBuilder.

File: jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java (hunk 1 above)
Comment (mcivantos-tribalyte) [reply jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: Done

File: jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java (hunk 2 above)
Comment (mcivantos-tribalyte) [reply jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: Done

File: jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java [hunk 4]
```diff
@@ -1328,6 +1327,33 @@ public void testUpdateTaskContentEvents() {
         Assertions.assertThat(taskEvents.get(5).getType()).isEqualTo(TaskEventType.COMPLETED);
         Assertions.assertThat(taskEvents.get(5).getUserId()).isEqualTo("Darth Vader");
     }
+    
+    @Test
+    public void testForwardTaskWithMsgEvents() {
+        Task task = new TaskFluent().setName("This is my task name")
+                .addPotentialGroup("Knights Templer")
+                .setAdminUser("Administrator")
```
Comment (domhanak) [inline jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: @mcivantos-tribalyte you should also explicitly add _Administrator_ as a potential owner, if you want to forward() this task. This should fix the permission exception.

Comment (domhanak) [reply jbpm-human-task/jbpm-human-task-audit/src/test/java/org/jbpm/services/task/audit/service/TaskAuditBaseTest.java]: to follow up I thought it not needed according to what is written in WS-HT spec, Administrator must be a potential owner if he wants to be able to forward

**Comment by MarianMacik** [comment]
jenkins retest this

**Pull request merged**